sdk/servicebus/service-bus/src/core/streamingReceiver.ts

```diff
@@ -314,7 +330,11 @@ export class StreamingReceiver extends MessageReceiver {
               bMessage.messageId,
               this.name
             );
-            this._onError!(translatedError);
+            this._onError!(translatedError, {
```

Comment:
Not related to this PR:

We abandon the message even if auto complete was set to false. Need to check if this is the expected behaviour or not. What if I wanted to deadletter or defer instead of abandon when I see certain error sources?

Reply:
In .NET, we call the error handler first, so if the user settles the message somehow, we wouldn't abandon it.

Reply:
@ramya-rao-a :

> We abandon the message even if auto complete was set to false. Need to check if this is the expected behaviour or not. 

Originally I was thinking "whoa, this is a bug" but thinking about it more it makes sense. The user isn't going to see this message again (it won't show up in any other callbacks, they need to re-request it from SB) so rather than let it sit there and wait for the full lock time to expire we can just abandon the lock instead. 

Considering we've had extensive design discussions around how the callback-based receivers are intended to be the simple path this fits in well.

> What if I wanted to deadletter or defer instead of abandon when I see certain error sources?

So technically they could - chronologically we do call processError before we do the auto-abandon logic. However, we don't pass in the message to the handler so they'd have to have somehow kept it around, know how to correlate the error with the message, etc... 

Ultimately they had a much easier path to accomplish this - they can just handle it in processMessage.

Reply:
> chronologically we do call processError before we do the auto-abandon logic.

Ok, so the user does have a chance to do something, but like you rightly pointed they wouldnt know which message is the error for